### ISSUE TITLE
reportUnnecessaryComparison false positive with Literal

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#4136]
**Describe the bug**
Pyright reports `Condition will always evaluate to False` where it should not. I believe it's not related to https://github.com/microsoft/pyright/issues/4107

**To Reproduce**
```python
L = Literal["a", "b"]


def f(x: L, y: L):
    if x != y:  # Condition will always evaluate to False
        ...
```

**Expected behavior**
Obviously x could be equal to y. So no error should be reported.

**VS Code extension or command-line**

pyright 1.1.278

**Comment by erictraut** [comment]
This is included in pyright 1.1.279, which I just published. It will also be included in a future version of pylance.
